lesson01/assignment/scripts.js

```diff
@@ -22,5 +35,48 @@ $(function () {
             input.val('');
         }
     });
+    submit2.on('click', function() {
+        const num = parseInt(input2.val());
+        //removed val variable because it wasn't being used anywhere else
+
+        if (!isNaN(num)) {
+            //checks to see if input value is a number
+            const answer = data.every((value) => value < num);
+            //check if there is a value that is under the input number
+            result2.text(answer);
+            input2.val('');
+        }
+    });
+    submit3.on('click', function() {
+        const num = parseInt(input3.val());
+        if (!isNaN(num)) {
+            const data2 = data.filter((value) => value < num);
+            //returns all values under input number
+            result3.text(data2);
+            input3.val('');
+        }
+    });
+    submit4.on('click', function() {
+        const num = parseInt(input4.val());
+        if (!isNaN(num)) {
+            const answer = data.includes(num);
+            //returns true or false depending if the input value is found wuthin the array
+            result4.text(answer);
+            input4.val('');
+        }
+    });
+    submit5.on('click', function() {
+        const num = parseInt(input5.val());
+        if (!isNaN(num)) {
+            const data2 = data.map((value) => value += num);
+            //adds the input value to each element in the array and then prints out into a seperate array.
+            result5.text(data2);
+            input5.val('');
+        }
+    });
 });
 
+/*
+concluding notes: I feel like there is a more efficient way of doing this rather than reusing a lot of the same functions and 
+using similarly named variables multiple times, but right now I can't really think of any ideas so I'll come back to this later.
```

Comment:
One way you can accomplish this is by passing an element into function, then accessing it's ID to determine which code to run.
```
<button type="button" onclick="myFunction(this)" id="submit1">Button1</button>
<button type="button" onclick="myFunction(this)" id="submit2">Button1</button>
<button type="button" onclick="myFunction(this)" id="submit3">Button1</button>

function myFunction(elem) {
    if(elem.id == "submit1")
    else if(elem.id == "submit2")
    else if(elem.id == "submit3")
}
```

See this stackoverflow article for other examples. You could also use a switch/case in this scenario. https://stackoverflow.com/questions/18410341/using-multiple-buttons-on-same-function-that-redirects-to-different-functions